Behavior policies added to the Policy Cookbook. (#25)
hivemq/hivemq-policy-cookbooks: behavior-validation/publish-quota/README.md

# Disconnect a client sending amount of messages outside of the specified range

> This cookbook is about to disconnect a client when the amount of published messages is outside the specified range.

Reviewer: ```suggestion
This cookbook is about disconnecting a client when the number of messages it has published is exceeds a specified limit.
```

### Use-case
> As a developer, I want to disconnect all clients not sending correct amount of messages.

For this use-case, a behavior policy is required.

### Policy

Consider the following behavior policy that disconnects a client when the amount of published messages is more than 10.

This policy uses `System.log` function to log explanatory messages on each state transition and `Mqtt.disconnect` function to disconnect a client if message count goest above 10.

`policy.json`:
```json
{
  "id": "max10",
  "matching": {
    "clientIdRegex": ".*"
  },
  "behavior": {
    "id": "Publish.quota",
    "arguments": {
      "minPublishes": 0,
      "maxPublishes": 10
    }
  },
  "onTransitions": [
    {
      "fromState": "Initial",
      "toState": "Connected",
      "Mqtt.OnInboundConnect": {
        "pipeline": [
          {
            "id": "logFunctionTransitionDebug",
            "functionId": "System.log",
            "arguments": {

>               "level": "INFO",

Reviewer: Also all "DEBUG"

      "toState": "Violated",
      "Mqtt.OnInboundPublish": {
        "pipeline": [
          {
            "id": "logFunction",
            "functionId": "System.log",
            "arguments": {
              "level": "WARN",
              "message": "Behavior policy ${policyId}: ${clientId} transitioned from ${fromState} to ${toState} on ${triggerEvent} at ${timestamp}"
            }
          },
          {
            "id": "logFunctionTransitionDebug",
            "functionId": "System.log",
            "arguments": {
              "level": "WARN",
              "message": "Ok, we had enough messages! ${clientId} will now be disconnected."
            }
          },
          {
            "id": "disconnectFunction",
            "functionId": "Mqtt.disconnect",
            "arguments": {}
          }
        ]
      }
    }
  ]
}
```

To upload `policy.json` to the broker, run the following command:

```bash
mqtt hivemq behavior-policy create --file policy.json
```

The policy is now applied and all incoming MQTT messages are subject to validation.

To delete the policy, run the following command:

```bash
mqtt hivemq behavior-policy delete --id max10
```

### Policy output

> Excessive amount of `System.log` calls used in the policy produces an output that facilitates understanding of the policy's behavior.

Reviewer: ```suggestion
The abundant use of `System.log` in the policy produces a log output that facilitates understanding of the policy's behavior.
```
